src/graph/GroupByExecutor.cpp

```diff
@@ -0,0 +1,435 @@
+                LOG(ERROR) << "Group `" << *groupName << "' isn't in output fields";
+                return Status::SyntaxError("Group `%s' isn't in output fields", groupName->c_str());
+            }
+            inputGroupCols.emplace(*groupName);
+            continue;
+        }
+
+        // Function call
+        if (it->expr()->isFunCallExpression()) {
+            continue;
+        }
+
+        // Check alias col
+        auto groupName = it->expr()->toString();
+        auto alisaIt = aliases_.find(groupName);
+        if (alisaIt != aliases_.end()) {
+            it = alisaIt->second;
+            auto gName = static_cast<InputPropertyExpression*>(it->expr())->prop();
+            if (it->expr()->isInputExpression()) {
+                inputGroupCols.emplace(*gName);
+            }
+            continue;
+        }
+        return Status::SyntaxError("Group `%s' isn't in output fields", groupName.c_str());
+    }
+
+    // Check yield cols
+    for (auto &it : yieldCols_) {
+        if (it->expr()->isInputExpression()) {
+            auto yieldName = static_cast<InputPropertyExpression*>(it->expr())->prop();
+            auto findIt = schemaMap_.find(*yieldName);
+            if (findIt == schemaMap_.end()) {
+                LOG(ERROR) << "Yield `" << *yieldName << "' isn't in output fields";
+                return Status::SyntaxError("Yield `%s' isn't in output fields", yieldName->c_str());
+            }
+            // Check input yield filed without agg fun and not in group cols
+            if (inputGroupCols.find(*yieldName) == inputGroupCols.end() &&
+                    it->getFunName().empty()) {
+                LOG(ERROR) << "Yield `" << *yieldName << "' isn't in group fields";
+                return Status::SyntaxError("Yield `%s' isn't in group fields", yieldName->c_str());
+            }
+        } else if (it->expr()->isVariableExpression()) {
+            LOG(ERROR) << "Can't support variableExpression: " << it->expr()->toString();
+            return Status::SyntaxError("Can't support variableExpression");
+        }
+        // TODO(laura): to check other expr
+    }
+    return Status::OK();
+}
+
+
+void GroupByExecutor::execute() {
+    FLOG_INFO("Executing Group by: %s", sentence_->toString().c_str());
+
+    if (rows_.empty()) {
+        DCHECK(onFinish_);
+        onFinish_();
+        return;
+    }
+
+    auto status = checkAll();
+    if (!status.ok()) {
+        DCHECK(onError_);
+        onError_(std::move(status));
+        return;
+    }
+
+    status = groupingData();
+    if (!status.ok()) {
+        DCHECK(onError_);
+        onError_(std::move(status));
+        return;
+    }
+
+    generateOutputSchema();
+
+    if (onResult_) {
+        auto outputs = setupInterimResult();
+        onResult_(std::move(outputs));
+    }
+
+    DCHECK(onFinish_);
+    onFinish_();
+}
+
+
+Status GroupByExecutor::groupingData() {
+    // key : group col vals, val: cal funptr
```

Comment:
Would you please make the annotation much more detailed and readable. We could use full word to explain the data structure here.